src/api/methods/getUnsignedTransferTx.js

```diff
@@ -0,0 +1,15 @@
+const makeTransfer = require('../../txHelpers/makeTransfer');
+
+module.exports = async (bridgeState, from, to, color, value) => {
+  const tx = await makeTransfer(
+    {
+      balances: bridgeState.currentState.balances,
+      unspent: bridgeState.currentState.unspent,
+    },
+    from,
+    to,
+    value,
+    color
+  );
+  return tx.toJSON();
```

Comment:
One thing, if we use this endpoint one light clients, then the light client still needs to properly encode it and sign it. Would it be better to return the `sigData` and have a light function to sign it and patch-in the signatures?

Reply:
@zyfrank what you think? :)

Reply:
This sounds good on paper, but it also means that the light client will be signing some random bytes no? Which means it could be signing anything. I think it should always reconstruct the tx first and have the user verify what they are signing.

Reply:
So basically you still need leap-core?

Reply:
A subset of it, I guess.

Reply:
Very small subset. Implementing signing function is much easier than implementing full transaction forming so I don't see an issue here. We can have separate bounties for writing a wallet frameworks for some of the most popular languages